ui/src/app/views/concept-homepage/component.ts

```diff
@@ -358,8 +358,8 @@ export class ConceptHomepageComponent implements OnInit {
 
   private filterConceptSelection(concepts: ConceptInfo[]) {
     concepts.forEach((concept) => {
-      concept.selected = this.conceptsToAdd.map(c => c.conceptId)
-        .indexOf(concept.conceptId) > -1;
+      const conceptSet = new Set(this.conceptsToAdd.map(c => c.conceptId));
```

Comment:
This is still n^2 if you build the set inside the for loop

Reply:
(move outside this callback)

Reply:
Haha - I knew that too. Yeah sorry about that. Fixed now